com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JdtUtils.java

```diff
@@ -112,31 +111,29 @@ public static IProject getProject(String projectName) {
     }
 
     /**
-     * Compute the possible source containers that the current project could be associated with.
+     * Compute the possible source containers that the specified project could be associated with.
      * <p>
-     * If current project is specified, try to parse the source containers from current project's classpath entries first,
-     * then the other projects at the same workspace.
+     * If the project name is specified, it will put the source containers parsed from the specified project's
+     * classpath entries in the front of the result, then the other projects at the same workspace.
      * </p>
      * <p>
-     * Otherwise, loop every projects at the current workspace and combine the parsed source containers directly.
+     * Otherwise, just loop every projects at the current workspace and combine the parsed source containers directly.
      * </p>
-     * @param currentProject
-     *                  the current project name.
+     * @param projectName
+     *                  the project name.
      * @return the possible source container list.
      */
-    public static ISourceContainer[] getSourceContainers(String currentProject) {
+    public static ISourceContainer[] getSourceContainers(String projectName) {
         Set<ISourceContainer> containers = new LinkedHashSet<>();
         List<IProject> projects = new ArrayList<>();
 
-        IProject targetProject = JdtUtils.getProject(currentProject);
+        IProject targetProject = JdtUtils.getProject(projectName);
         if (targetProject != null) {
             projects.add(targetProject);
         }
 
         List<IProject> workspaceProjects = Arrays.asList(ResourcesPlugin.getWorkspace().getRoot().getProjects());
-        projects.addAll(workspaceProjects.stream().filter((project) -> {
-            return !project.equals(targetProject);
-        }).collect(Collectors.toList()));
+        projects.addAll(workspaceProjects);
```

Comment:
There can be duplicated entries inside the `projects` collection. The `targetProject` is already inside.
```
ResourcesPlugin.getWorkspace().getRoot().getProjects()
JdtUtils.getProject(projectName)
```
Does the return value of the first call contain the return value of the second call?

Reply:
YES, it contains the second result.

Reply:
But it's OK to have duplicated projects because the later computation code has logic to avoid repeated calculation on the duplicated classpath entries.